File: modSrc/common-1.20.1/src/main/java/io/github/summpot/beaconauth/login/LoginQueryType.java
package io.github.summpot.beaconauth.login;

import net.minecraft.resources.ResourceLocation;
import io.github.summpot.beaconauth.BeaconAuthMod;

/**
 * Identifiers for BeaconAuth login-phase custom query packets.
 */
public enum LoginQueryType {
    PROBE("probe"),
    INIT("init"),
    LOGIN_URL("login_url"),
    VERIFY("verify");

    private final ResourceLocation id;

    LoginQueryType(String path) {
        this.id = new ResourceLocation(BeaconAuthMod.MOD_ID, path);
    }

    public ResourceLocation id() {
        return id;
    }
}

File: modSrc/common-1.20.1/src/versioned/kotlin/io/github/summpot/beaconauth/login/LoginQueryType.kt
package io.github.summpot.beaconauth.login

import io.github.summpot.beaconauth.BeaconAuthMod
import net.minecraft.resources.ResourceLocation

/**
 * Identifiers for BeaconAuth login-phase custom query packets (1.20.1).
 */
enum class LoginQueryType(private val path: String) {
    PROBE("probe"),
    INIT("init"),
    LOGIN_URL("login_url"),
    VERIFY("verify");

    fun id(): ResourceLocation = ResourceLocation(BeaconAuthMod.MOD_ID, path)
}
